Lf improve annotations candidate generation

Luis-R-Flores opened the pull request:
## What does this change?

Overwriting outdated versions of the following jupyter notebooks:

- kb_datasets/merge_and_clean_kb_datasets.ipynb
- entity_source_data/resample_gu_articles.ipynb
- el_notebook_20220929.ipynb
- ec2_instance_code/2022_10_03/open_sanctions.ipynb 

Main additions are to el_recipe.py, the python file running the EL prodigy annotation task. Main changes are the addition of code to improve the KB entity candidate generation.   
 
## How to test

Does the prodigy recipe launch and work as expected? Do relevant candidates get surfaced for single names? (e.g. Trump mention -> KB Donald Trump, the former president). Do mentions with multiple names work as well as before?

Luis-R-Flores committed:
Clean kb further and stricter resample of gu articles. Progigy recipe now in good condition for BWH. Added gu trf model.

Luis-R-Flores committed:
Clean kb further and stricter resample of gu articles. Progigy recipe now in good condition for BWH. Added gu trf model.

Luis-R-Flores committed:
Fix position abbreviation. Fix "name is nonsense" issue. Add gu model pipeline to nel notebook. Clean up prodigy recipe.

Luis-R-Flores committed:
Minor data quality issues. Further standardise kb names. Adjust Prodigy UI printing lenght and add missing full stop marks.

Luis-R-Flores committed:
Started work on improving candidate generation

Luis-R-Flores committed:
Merge branch 'ms/prodigy-recipe' into lf_jai_data_cleaning_and_prodigy_recipe

Luis-R-Flores committed:
Improve speed of candidate generation by replacing embedding with BoWs

Luis-R-Flores committed:
Merge branch 'master' of github.com:guardian/jai_2022_sandbox

Luis-R-Flores committed:
merge feature branch changes to copy of master

Luis-R-Flores committed:
Integrate improvements to candidate generation

michel-ds commented:
This now runs live, shall we merge, @Luis-R-Flores ?

Luis-R-Flores committed:
fix empty span issue causing recipe to stop early

michel-ds reviewed (commented):
Looks promising.

Can you please remove the change to `training/train.ipynb` as this is out of scope for this PR?

Luis-R-Flores committed:
Remove model training notebook

michel-ds committed:
Fix: Add quotation mark standardisation

This fixes the NER model incorrectly flagging up unicode quotation marks as [PERSON]
